device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceClientManager.java

```diff
@@ -53,7 +53,7 @@ public void run()
                             System.exit(-1);
                         }
 
-                        if (newStatusReason == IotHubConnectionStatusChangeReason.EXPIRED_SAS_TOKEN)
+                        if (newStatus == IotHubConnectionStatus.DISCONNECTED && newStatusReason == IotHubConnectionStatusChangeReason.EXPIRED_SAS_TOKEN)
```

Comment:
MQTT connections will routinely get DISCONNECTED_RETRYING with reason EXPIRED_SAS_TOKEN, so no need to end the sample unless it is DISCONNECTED